src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml

```diff
@@ -10,13 +10,14 @@
         <service id="controller_name_converter" class="Symfony\Bundle\FrameworkBundle\Controller\ControllerNameParser">
             <tag name="monolog.logger" channel="request" />
             <argument type="service" id="kernel" />
+            <deprecated>The "%service_id%" service is deprecated since Symfony 4.4.</deprecated>
         </service>
 
         <service id="controller_resolver" class="Symfony\Bundle\FrameworkBundle\Controller\ControllerResolver">
             <tag name="monolog.logger" channel="request" />
             <argument type="service" id="service_container" />
-            <argument type="service" id="controller_name_converter" />
```

Comment:
what if someone replaced the `controller_name_converter` service? That will break their whole app. To me that is a bigger BC break in 4.4 than removing a constructor argument in master.

Reply:
Ok, updated to what I think is a sensible tradeoff:
The service is not deprecated anymore at the DI level so we can still inject it in core dependent classes. The corresponding constructor argument in dependent classes has been made implicit so that they can detect non-core injections and trigger. 
Also added an UPGRADE entry for services that are not officially marked as deprecated but will be removed in 5.0